### ISSUE TITLE
scVIVA: early stopping works as intended?

### ISSUE THREAD

**Issue by dbdimitrov** [scverse/scvi-tools#3421]
Hi,

I've been running scVIVA on some cosmx data and I've noticed that if I use early stopping, it would essentially always stop at 1 + `early_stopping_patience`. 

I'm currently travelling and I don't have the time to sit down and check in detail, but it's likely that the (negative) sign of the loss is leading to it stopping prematurely.

Might be a good idea to check.

Daniel

**Comment by dbdimitrov** [comment]
PS. I'm running this on the latest version of scvi-tools: i.e. `1.3.2`

**Comment by canergen** [comment]
Thanks. I assume you used elbo_validation loss as the criterion for early stopping? @LevyNat I think we did all experiments without early stopping and the issue might be that additional losses tend to let ELBO increase over time. Easiest suggestion we should just check for it in model.train and throw an error (so early_stopping not supported in scVIVA).

**Comment by dbdimitrov** [comment]
Hi @canergen,

Thanks for the prompt response. Yes, I was using elbo_validation as criterion.

From what I'm getting, I should avoid using early stopping for now, but please do let me know if there is a val loss that would work better than the ELBO :) 

I'm mostly asking since I'm doing some method comparisons and trying to give each method it's best chance to do well.

**Comment by canergen** [comment]
Looking forward to the benchmark. Feel free to contact us with issues (errors or performance issues) that arise using scVIVA. We are happy to improve tools based on solid benchmarks.

**Comment by LevyNat** [comment]
Hi, @canergen it is true that the current criterion is the elbo without the niche losses. It means that at some point, the elbo might stop decreasing and increase a bit. But if @dbdimitrov says that the training stops at epoch 1+patience, it means that the elbo does not decrease at the beginning of the training, which is not expected. I'd be happy to debug this with you, best thing is to start by using the same parameters as in the tutorial (https://docs.scvi-tools.org/en/stable/tutorials/notebooks/spatial/scVIVA_tutorial.html) and plot the losses, then we'll see what goes wrong. You can send me privately

**Comment by dbdimitrov** [comment]
Hi @LevyNat,

I changed a bit the parameters and specifically halving the default learning rate (as in the tutorial) seems to fix it. 

I thought that not negating the ELBO (or the reported loss) might've been the reason - it was a bit suspicious that it was stopping exactly at the patience threshold, while e.g. SCANVI took much longer to early stop with essentailly the same hyperparams.

Thanks for the responses!
